### DISCUSSION TITLE
Referencing another file within a function

### DISCUSSION THREAD

**Discussion by dmorda** [gatsbyjs/gatsby#33243] [Help] [answered]
I have a Gatsby function that sends email (api/sendmail/sendmail.ts). When it executes, it needs to grab an EJS template file (api/sendmail/email_templates/template.ejs) to create the HTML email output.

Included below are a few pieces of the Gatsby function to illustrate how I'm trying to do it. It works when running Gatsby locally, but when I push it to Netlify it can't find the file.

```
// api/sendmail/sendmail.ts

// I can't use ./email_templates/template.ejs so I found this tricks which works locally. */
const __filename = fileURLToPath(import.meta.url)
const __dirname = path.dirname(__filename)

[...bunch of other code]

const templateFile = path.join(__dirname, `/email_templates/${formName}.ejs`)

ejs.renderFile(templateFile, { data })
      
```

Is there a trick to referencing a file within a Gatsby function?

**Comment by dmorda** [comment]
I also posted something similar on Netlify forums.

https://answers.netlify.com/t/do-gatsby-functions-work-with-the-included-files-property/44363

**Comment by KyleAMathews** [comment, marked as answer]
Gatsby functions only bundle files that you require. So instead of using `ejs.renderFile`, require the template and compile it and then use `ejs.render`.

**Reply by dmorda** [reply]
I think I'm following you, but just to clarify, are you saying to do something like this?

```
template.js

const myTemplate = () => {
  return `<html><body><%- message %></body></html>`
}

export default myTemplate
```

And then in my sendmail, doing something like this?

```
sendmail.js

import myTemplate from './template'

[...]

ejs.render(myTemplate, message);

[..]
```

**Reply by KyleAMathews** [reply]
Yes or you can use the raw-loader to directly import the template like:

`import templateStr from 'raw-loader!./template.ejs';`

**Reply by dmorda** [reply]
Whoa, I didn't know that raw-loader trick. You just saved the day, thanks @KyleAMathews!
